### ISSUE TITLE
Invalid warning `TypeVar "T" appears only once in generic function signature`

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#2509]
With this code
```python
T = TypeVar('T')

class NamedThing(Generic[T]):
    def __init__(self, thing: T, name: str):
        self.thing: T = thing
        self.name: str = name


def get_name(named_thing: NamedThing[T]) -> str:
    return named_thing.name
```

I get the following warning
> TypeVar "T" appears only once in generic function signature

but if I remove the TypeVar in `get_name` I get this instead
> Expected type arguments for generic class "NamedThing"

I am using Pyright v1.1.182  in VS Code

**Comment by nicholas-johnson-techxcel** [comment]
Since the documentation says that reportInvalidTypeVarUse defaults to warning, why is it erroring me?

**Comment by nicholas-johnson-techxcel** [comment]
Something should never be an error if there are legitimate cases for it. You can error it out in the illegitimate cases. But this seems to be "we could not decide so we just tossed a coin" based on the commentary. Plus, the documentation says it is a warning by default. If it has been escalated by `"python.analysis.typeCheckingMode": "strict",` then that needs to be fixed because this is not something we enable to turn warnings into errors. It is something we use to show all the actual errors.
